### PR TITLE
[Lint] Add black and pylint 

### PR DESCRIPTION
**Pull request by Herklos**
Depends on https://github.com/Drakkar-Software/OctoBot-Channels/pull/25
Depends on https://github.com/Drakkar-Software/OctoBot-Commons/pull/29

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,6 +37,14 @@ matrix:
         - rm -rf $PACKAGE_FOLDER
         - pytest tests
 
+    - name: "Linux - Python 3.8-dev - Lint"
+        stage: test
+        os: linux
+        language: python
+        script:
+          - black $PACKAGE_FOLDER --diff --check
+          -
```
Comment (GuillaumeDSM) [inline .travis.yml]: empty **-** ?

File: octobot_channels/consumer.py [hunk 2]
```diff
@@ -38,17 +41,11 @@ def __init__(self, callback: object, size: int = 0, filter_size: bool = False):
 
         """
         Should be used as the perform while loop condition
-            while(self.should_stop):
-                ...
+            >>> while(self.should_stop):
```
Comment (GuillaumeDSM) [inline octobot_channels/consumer.py:44]: **>>>** ?

File: .travis.yml (hunk 1 above)
Comment (Herklos) [reply .travis.yml]: It's WIP, will be `pylint` soon

File: octobot_channels/consumer.py (hunk 2 above)
Comment (Herklos) [reply octobot_channels/consumer.py:44]: https://www.sphinx-doc.org/en/1.5/markup/code.html

File: .gitignore [hunk 3]
```diff
@@ -103,9 +103,13 @@ venv.bak/
...
+# ide
 .idea
 
+# cython
 *.html
 *.c
-
 cython_debug/
+
+# doc
```
Comment (GuillaumeDSM) [inline .gitignore]: 👍

File: .travis.yml [hunk 4]
```diff
@@ -37,16 +37,24 @@ matrix:
         - rm -rf $PACKAGE_FOLDER
         - pytest tests
 
+    - name: "Linux - Python 3.8-dev - Lint"
```
Comment (GuillaumeDSM) [inline .travis.yml:40]: 👍

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.3.22] - 2020-03-26
+### Added
+- Documentation basis with sphinx
+- Pylint check on CI
+- Black check on CI
```
Comment (GuillaumeDSM) [inline CHANGELOG.md:11]: 💯

File: octobot_channels/channels/channel.py [hunk 6]
```diff
@@ -20,15 +20,16 @@
...
-To access to channels a 'Channels' singleton is created to manage instances. 
-"""
 
+# pylint: disable=undefined-variable, not-callable
+class Channel:
+    """
+    A Channel is the object to connect a producer / producers class(es) to a consumer / consumers class(es)
+    It contains a registered consumers dict to notify every consumer when a producer 'send' something.
+    It contains a registered producers list to allow producer modification through 'modify'.
+    To access to channels a 'Channels' singleton is created to manage instances.
```
Comment (GuillaumeDSM) [inline octobot_channels/channels/channel.py]: 💯 
To access channels  (no "to" required here)

File: octobot_channels/consumer.py [hunk 7]
```diff
@@ -18,13 +18,16 @@
...
-"""
-
 
 class Consumer:
-    def __init__(self, callback: object, size: int = 0, filter_size: bool = False):
+    """
+    A consumer keeps reading from the channel and processes any data passed to it.
+    A consumer will start consuming by calling its 'consume' method.
+    The data processing implementation is coded in the 'perform' method.
+    A consumer also responds to channel events like pause and stop.
```
Comment (GuillaumeDSM) [inline octobot_channels/consumer.py:27]: 👍

File: octobot_channels/consumer.py [hunk 8]
```diff
@@ -116,16 +116,35 @@ def __str__(self):
 
 
 class InternalConsumer(Consumer):
+    """
+    An InternalConsumer is used to loop internally of the class with a callback
```
Comment (GuillaumeDSM) [inline octobot_channels/consumer.py]: Could you rephrase this one ? It's kinda hard to understand

**Review by GuillaumeDSM — COMMENTED** [review]
great ! 🚀

File: octobot_channels/consumer.py [hunk 9]
```diff
@@ -116,16 +116,35 @@ def __str__(self):
 
 
 class InternalConsumer(Consumer):
+    """
+    An InternalConsumer is an classic Consumer except that his callback is declared internally
```
Comment (GuillaumeDSM) [inline octobot_channels/consumer.py]: is **a** classic Consumer then perfect 💯

**Review by GuillaumeDSM — APPROVED** [review]
